### DISCUSSION TITLE
v2.4.1

### DISCUSSION THREAD

**Discussion by mikecao** [umami-software/umami#2177] [General]
## Fixes

- Fixed filters not working on overview page
- Fixed event data not returning records
- Fixed MySQL event data query error
- Fixed wrong HTTP response for metrics endpoint

<hr /><em>This discussion was created from the release <a href='https://github.com/umami-software/umami/releases/tag/v2.4.1'>v2.4.1</a>.</em>

**Comment by skipperbent** [comment]
Confirms the events page works in this latest update.

But again I'm confused about the idea behind the events page. I can see multiple of the same events appearing; for instance like appears 4 times - 3 of them with the same field value. They all take me to the same page, which is a list filled with data that i have no idea how to use. No timestamp, nothing is grouped together. I would expect to see Like => id: x, type: y.

Im genuinely curious how this is supposed to work or be used? I mean it works now as it displays the values, but now the data is just all over the place with no identifier, date, filtering, sorting or anything?

**Reply by mikecao** [reply]
Let's say you have an event called `checkout` with the following fields, `cart_items` and `has_coupon`. When you bring up the event data page will see:

| Event    | Field      | Total records |
|----------|------------|---------------|
| checkout | cart_items | 100           |
| checkout | has_coupon | 25            |

From this you can deduce that 25% of cart checkouts used a coupon. Now you click on `checkout` so you can see the exact values and see something like this:

| Field      | Value | Total records |
|------------|-------|---------------|
| cart_items | 3     | 50            |
| cart_items | 2     | 25            |
| cart_items | 1     | 25            |
| has_coupon | true  | 25            |

From this you see that 50% of carts had 3 items and 25% had either 2 items or 1 item.

So it really depends on how you define your event data. In the future we will provide ways for you to browse individual events and entire event data objects.

**Reply by skipperbent** [reply]
It feels a bit convoluted just like the way users are managed.

So I have multiple "like" events, with fields type and id - but I can't tell which is which only for the events that serve one purpose.

In my brain the list would show all my events but never the same events twice, then I click and get event-data grouped by event fields with a counter if the same data appears multiple times.

![Screenshot_20230803-020819](https://github.com/umami-software/umami/assets/634762/a57f6ef9-1daf-4b59-9593-89515eed35f7)

I can see what IDs are like, but not the type which is part of the event data. And it's real problematic as id is not necessarily unique, a comment like can have the same id as a track like. So I would have to change my events data to id: 'comment_5' otherwise the dashboard would see them as the same.

![Screenshot_20230803-021716](https://github.com/umami-software/umami/assets/634762/06fc4da4-f263-4834-bf48-1bdeac453d8a)

**Reply by mikecao** [reply]
You definitely should not see multiple events with the same field like that. Possibly a bug. But yes, this summary view may be only useful in certain cases. We will be exposing more ways to view your event data in the future. As always, we welcome any suggestions.

**Reply by skipperbent** [reply]
Thanks for clarifying I appreciate the detailed response and very nice job with the project so far, impressive stuff.

**Comment by blyamur** [comment]
Friends, tell me please, such a problem.
In the latest 2-3 versions, only the main page / . Previously, it worked and displayed all pages. Deleted recreated and hung up the code in different places. The result is the same ((

But in the "more" section in the URL Title tab, the url is only / and the titles are different
